From dafa48600a197c1e8aae028a000057a50b985fa6 Mon Sep 17 00:00:00 2001
From: Neil Horman <nhorman@redhat.com>
Date: Wed, 22 Oct 2014 18:03:07 -0400
Subject: [net] usb/cdc_ncm: remove non-standard NCM device IDs

Message-id: <1414000989-9522-4-git-send-email-nhorman@redhat.com>
Patchwork-id: 98073
O-Subject: [RHEL7.1 PATCH v2 3/5] net: cdc_ncm: remove non-standard NCM device IDs
Bugzilla: 1129796
RH-Acked-by: Don Zickus <dzickus@redhat.com>
RH-Acked-by: Stefan Assmann <sassmann@redhat.com>
RH-Acked-by: Dean Nelson <dnelson@redhat.com>

Remove device IDs of NCM-like (but not NCM-conformant) devices, that are
handled by the huawwei_cdc_ncm driver now.

Signed-off-by: Enrico Mioso <mrkiko.rs@gmail.com>
Signed-off-by: Bjorn Mork <bjorn@mork.no>
Signed-off-by: David S. Miller <davem@davemloft.net>
(cherry picked from commit 9fea037de5f33973d48e8f893a0087c6f9861e7b)

BZ: 1129796
Brew: http://brewweb.devel.redhat.com/brew/taskinfo?taskID=8150596
Testing: Tested by reporter with success
Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/net/usb/cdc_ncm.c b/drivers/net/usb/cdc_ncm.c
index 555d3ec..5f02400 100644
--- a/drivers/net/usb/cdc_ncm.c
+++ b/drivers/net/usb/cdc_ncm.c
@@ -1231,17 +1231,6 @@ static const struct usb_device_id cdc_devs[] = {
    .driver_info = (unsigned long)&wwan_info,
  },
 
- /* Huawei NCM devices disguised as vendor specific */
- { USB_VENDOR_AND_INTERFACE_INFO(0x12d1, 0xff, 0x02, 0x16),
-   .driver_info = (unsigned long)&wwan_info,
- },
- { USB_VENDOR_AND_INTERFACE_INFO(0x12d1, 0xff, 0x02, 0x46),
-   .driver_info = (unsigned long)&wwan_info,
- },
- { USB_VENDOR_AND_INTERFACE_INFO(0x12d1, 0xff, 0x02, 0x76),
-   .driver_info = (unsigned long)&wwan_info,
- },
-
  /* Infineon(now Intel) HSPA Modem platform */
  { USB_DEVICE_AND_INTERFACE_INFO(0x1519, 0x0443,
   USB_CLASS_COMM,
-- 
1.7.1